rts/Game/UI/SelectionKeyHandler.cpp

```diff
@@ -383,6 +379,16 @@ void CSelectionKeyHandler::DoSelection(std::string selectString)
 			filter = ReadToken(selectString);
 		}
 
+		// We have to use this ad-hoc logic for handling OR-able IdMatches filtering
+		if (filter == "IdMatches") {
+			ReadDelimiter(selectString);
+
+			auto& matchSet = _not ? idNotMatchesSet : idMatchesSet;
```

Comment:
would it be possible to use `if (filter == "IdMatches" && not _not)` to avoid the `idNotMatchesSet` var?

Reply:
Sure, but how would we handle the `_Not_IdMatches` case?

Reply:
doesn't it keep the previous behaviour of AND if we don't handle it in this new block?

Reply:
We'd have to restore the `IdMatches` filter singleton, it's a possibility. As someone having to deal with this code, would you prefer to have the whole IdMatches thing in its encapsulated logic or would you be fine with having IdMatches singleton that only works with _Not + the ad hoc case?

Reply:
Good call!

Reply:
I think overall it would be good to keep explicit code only for cases that differ from the default behaviour (i.e. handle OR explicitly and let AND be done via the singleton). I don't know though, your call.